### PR TITLE
[LRUCache] Expose LRU Cache status to metrics

### PR DESCRIPTION
**Pull request by acelyc111**
## Proposed changes

Expose LRU Cache status to metrics would be helpful to diagnose
problems like high usage, low hit rate.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

**Comment by morningman** [comment]
Hi @acelyc111 , thanks for your PR.
And I suggest to add document for new API. Better to place it here:
`administrator-guide/http-actions/be/lru-cache-action.md`.

You can refer to the content format in `administrator-guide/http-actions/fe/connection-action.md`

**Comment by acelyc111** [comment]
Please review https://github.com/apache/incubator-doris/pull/4751 at first

**Comment by acelyc111** [comment]
> Hi @acelyc111 , thanks for your PR.
> And I suggest to add document for new API. Better to place it here:
> `administrator-guide/http-actions/be/lru-cache-action.md`.
> 
> You can refer to the content format in `administrator-guide/http-actions/fe/connection-action.md`

@morningman I've change to expose these status to metric instead of http API, metric is more firendly to third-party monitor system.

File: be/src/olap/lru_cache.cpp [hunk 1]
```diff
@@ -420,8 +425,7 @@ Slice ShardedLRUCache::value_slice(Handle* handle) {
 }
 
 uint64_t ShardedLRUCache::new_id() {
-    MutexLock l(&_id_mutex);
-    return ++(_last_id);
+    return _last_id.fetch_add(1, std::memory_order_relaxed);;
```
Comment (morningman) [inline be/src/olap/lru_cache.cpp:428]:
```suggestion
    return _last_id.fetch_add(1, std::memory_order_relaxed);
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**